Source/Core/Core/IOS/Network/KD/NWC24DL.cpp

```diff
@@ -58,10 +58,9 @@ void NWC24Dl::WriteDlList() const
     ERROR_LOG_FMT(IOS_WC24, "Failed to open or write WC24 DL list file");
 }
 
-void NWC24Dl::ResetConfig()
+bool NWC24Dl::DoesEntryExist(u16 entry_index)
 {
-  // m_fs->Delete(PID_KD, PID_KD, CONFIG_PATH);
-  // TODO: implement
+  return m_data.entries[entry_index].low_title_id != 0;
```

Comment:
This (plus a few other functions here) should probably check that `entry_index < MAX_ENTRIES`.

Reply:
Actually looking at the call dependencies, the other functions only happen if this one succeeds, so this is the only critical one. An `ASSERT()` in the other ones wouldn't hurt though.

Reply:
Since `entry_index` is from `NetKDRequestDevice::HandleNWC24DownloadNowEx`, wouldn't it be better to `ASSERT()` there before we start calling anything?

Reply:
Well, yes, you can check it there, but an assert is not the right choice there since the value comes directly from the emulated system's memory and could be gibberish. Asserting is for checking something that should always be true and indicates a programming error if it's not.

So if you want to check it there, you should probably do something like this before the call to `DoesEntryExist()`:

```
  if (entry_index >= NWC24::NWC24Dl::MAX_ENTRIES)
  {
    ERROR_LOG_FMT(IOS_WC24, "NET_KD_REQ: Entry index out of range.");
    WriteReturnValue(NWC24::WC24_ERR_BROKEN, request.buffer_out);
    return IPCReply(NWC24::WC24_ERR_BROKEN);
  }
```